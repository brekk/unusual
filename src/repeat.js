import { curry } from 'katsu-curry'

export const repeat = curry((total, fn) =>
  Array(total)
    .fill(0)
    .map((_, i) => fn(i))
)
