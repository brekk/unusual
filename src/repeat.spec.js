import { repeat } from './repeat'

test('repeat', () => {
  expect(repeat(10, i => (i % 2 === 0 ? 'even' : 'crooked'))).toEqual([
    'even',
    'crooked',
    'even',
    'crooked',
    'even',
    'crooked',
    'even',
    'crooked',
    'even',
    'crooked',
  ])
})
