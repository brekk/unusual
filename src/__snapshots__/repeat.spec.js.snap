// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unusual.repeat 1`] = `
[
  "d",
  "3",
  "3",
  "1",
  "f",
  "e",
  "a",
  "a",
  "d",
  "8",
]
`;
